kt/godot-runtime/src/main/kotlin/godot/core/NativeCoreTypeWeakReference.kt

```diff
@@ -0,0 +1,8 @@
+package godot.core
+
+import java.lang.ref.WeakReference
+
+internal class NativeCoreTypeWeakReference(referent: NativeCoreType)
+    : WeakReference<NativeCoreType>(referent) {
+    val coreVariantType = referent.coreVariantType

```

Comment:
doesn't this create a normal reference to `coreVariantType` and thus the ref to `referent` isn't weak anymore?
Souldn't it be (pseudo code): 
```kotlin
val coreVariantType: VariantType
    get() = referent.coreVariantType
```

Reply:
It's a normal reference to coreVariantType yes, but this has no effect on referent.